### PR TITLE
Clarify that conversion tactics will succeed even if the context doesn't change

### PR DESCRIPTION
**Pull request by jfehrle**

### CODE REVIEW

**Comment by jfehrle** [comment]
@coq/doc-maintainers Would someone review, please?  Thanks.

File: doc/sphinx/proofs/automatic-tactics/auto.rst [hunk 1]
```diff
@@ -412,15 +412,15 @@ Creating Hints
...
-      The :cmd:`Hint Rewrite` now supports locality attributes like other `Hint` commands.
-
    .. versionchanged:: 8.18
 
       The default value for hint locality outside sections is
       now :attr:`export`. It used to be :attr:`global`.
 
+   **Usage tip**: tactics that can succeed even if they don't change the context, such as
+   most of the :ref:`conversion tactics <applying-conversion-rules>`, should
+   be prefaced with :tacn:`progress` to avoid needless repetition of the tactic.
```
Comment (SkySkimmer) [inline doc/sphinx/proofs/automatic-tactics/auto.rst:422]: this note seems like it should be on Hint Extern specifically, not the general Hint preamble

Comment (jfehrle) [reply doc/sphinx/proofs/automatic-tactics/auto.rst]: Good point.  Updated.  If no further comments, would you care to merge?

**Review by SkySkimmer — APPROVED** [review]

File: doc/sphinx/proofs/automatic-tactics/auto.rst [hunk 2]
```diff
@@ -526,11 +522,13 @@ Creating Hints
       :tacn:`unfold`. :n:`@natural` is the cost, :n:`@one_pattern` is the pattern
       to match and :n:`@ltac_expr` is the action to apply.
 
-      .. note::
+      **Usage tip**: tactics that can succeed even if they don't change the context,
+      such as most of the :ref:`conversion tactics <applying-conversion-rules>`, should
```
Comment (SkySkimmer) [inline doc/sphinx/proofs/automatic-tactics/auto.rst:526]:
```suggestion
      such as most of the :ref:`conversion tactics <applyingconversionrules>`, should
```

Comment (jfehrle) [reply doc/sphinx/proofs/automatic-tactics/auto.rst:526]: Got it, thanks!

**Review by Zimmi48 — APPROVED** [review]

**Comment by SkySkimmer** [comment]
@coqbot merge now

**Pull request merged**